### PR TITLE
notification

### PR DESCRIPTION
**Pull request by ThuyenPVT**
making new friend : https://gyazo.com/0633f63d2cd1effbde1b352eafb4d31f

### CODE REVIEW

File: android/app/src/main/kotlin/com/chuongdever/structure_flutter/MainActivity.kt [hunk 1]
```diff
@@ -3,4 +3,5 @@ package com.chuongdever.structure_flutter
 import io.flutter.embedding.android.FlutterActivity
 
 class MainActivity : FlutterActivity() {
-}
+
+}
```
Comment (chuongdevJC) [inline android/app/src/main/kotlin/com/chuongdever/structure_flutter/MainActivity.kt]: .

File: lib/bloc/blocs/friend_bloc.dart [hunk 2]
```diff
@@ -1,25 +1,63 @@
...
+        accept: false,
+      );
+
+      await _friendRepository.makingFriends(
+        currentID: recipientID,
+        recipientID: currentUserID,
+        name: currentUserName,
+        pending: true,
+        accept: false,
+      );
```
Comment (chuongdevJC) [inline lib/bloc/blocs/friend_bloc.dart]: duplicate func

File: lib/data/entities/message.dart [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+}
+
+class Message {
+  final String senderID;
+  final String content;
+  final Timestamp timestamp;
+  final MessageType type;
+
+  Message({this.senderID, this.content, this.timestamp, this.type});
+}
```
Comment (chuongdevJC) [inline lib/data/entities/message.dart]: .

File: lib/data/source/remote/config/config.dart [hunk 4]
```diff
@@ -1 +1,127 @@
...
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+
+// ignore_for_file: non_constant_identifier_names
+// ignore_for_file: camel_case_types
+// ignore_for_file: prefer_single_quotes
+
+// This file is automatically generated. DO NOT EDIT, all your changes would be lost.
+class S implements WidgetsLocalizations {
```
Comment (chuongdevJC) [inline lib/data/source/remote/config/config.dart]: .

File: lib/pages/friend_list/widgets/friend_profile.dart [hunk 5]
```diff
@@ -33,27 +79,107 @@ class FriendProfile extends StatelessWidget {
...
-                    shape: RoundedRectangleBorder(
-                      borderRadius: BorderRadius.circular(30.0),
-                    ),
-                    onPressed: onPressed,
-                    child: isActiveButton
-                        ? Text('Send Request', style: AppStyles.white_10)
-                        : Text('Pending', style: AppStyles.white_10),
-                  ),
-                ),
+                    child: GestureDetector(
```
Comment (chuongdevJC) [inline lib/pages/friend_list/widgets/friend_profile.dart]: so long function. please extract it

File: lib/pages/user_information/user_information_screen.dart [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+
+  @override
+  _UserInformationScreenState createState() => _UserInformationScreenState();
+}
+
+class _UserInformationScreenState extends State<UserInformationScreen> {
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      color: Colors.white,
```
Comment (chuongdevJC) [inline lib/pages/user_information/user_information_screen.dart]: remove if unused

**Review by chuongdevJC — APPROVED** [review]

File: lib/bloc/blocs/register_bloc.dart [hunk 7]
```diff
@@ -42,8 +41,13 @@ class RegisterBloc extends Bloc<RegisterEvent, RegisterState> {
     yield RegisterState.loading();
     try {
       final String uid = await _userRepository.signUp(email, password);
+      //upload avatar image file to storage
       var result = _storageRepository.uploadUserImage(uid, imageURL);
+      //get avatar image url link after image file
       var _imageURL = await (await result).ref.getDownloadURL();
+      //update name and image link into User collection
+      await _userRepository.updateUserInfo(name, _imageURL);
+      //create new user
```
Comment (truyenjc) [inline lib/bloc/blocs/register_bloc.dart]: remove comments, this block code easy to understand, function name already explain what its do
https://refactoring.guru/smells/comments

File: lib/data/entities/account.dart [hunk 8]
```diff
@@ -1,5 +1,5 @@
 import 'package:cloud_firestore/cloud_firestore.dart';
-
+import 'package:intl/intl.dart';
```
Comment (truyenjc) [inline lib/data/entities/account.dart]: remove unused import

File: lib/data/source/remote/user_remote_datasource.dart [hunk 9]
```diff
@@ -14,18 +14,19 @@ abstract class UserRemoteDataSource {
...
 
-  UserRemoteDataSourceImpl({
-    FirebaseAuth firebaseAuth,
-    GoogleSignIn googleSignIn,
-  })  : _firebaseAuth = firebaseAuth ?? FirebaseAuth.instance,
-        _googleSignIn = googleSignIn ?? GoogleSignIn();
+  UserRemoteDataSourceImpl(
+    this._firebaseAuth,
+    this._googleSignIn,
+  );
```
Comment (truyenjc) [inline lib/data/source/remote/user_remote_datasource.dart]:
```suggestion
  UserRemoteDataSourceImpl(
    this.firebaseAuth,
    this.googleSignIn,
  );
```

File: lib/pages/authen/register/widgets/app_bar.dart [hunk 10]
```diff
@@ -6,18 +6,19 @@ import 'package:structure_flutter/pages/authen/login/login_page.dart';
 
 Widget appBar(BuildContext context) {
```
Comment (truyenjc) [inline lib/pages/authen/register/widgets/app_bar.dart]: should be define widget class

File: lib/pages/friend_list/friend_list_page.dart [hunk 11]
```diff
@@ -1,76 +1,84 @@
...
 
 class _MessageState extends State<FriendListPage> {
-  CollectionReference users = FirebaseFirestore.instance.collection('Users');
-
-  final _accountRepository = getIt<AccountRepository>();
-
   final _snackBar = getIt<SnackBarWidget>();
 
-  final _loadDataBloc = getIt<FriendBloc>();
+  final _friendDataSource = getIt<FriendRemoteDataSource>();
```
Comment (truyenjc) [inline lib/pages/friend_list/friend_list_page.dart]: remove (kill)

File: lib/pages/friend_list/friend_list_page.dart [hunk 12]
```diff
@@ -106,37 +111,40 @@ class _MessageState extends State<FriendListPage> {
...
+      currentUserName,
+      recipientID,
+      recipientName,
+    ));
+    // _friendDataSource.updateMakingFriends(
+    //   currentID: currentUserID,
+    //   recipientID: recipientID,
+    //   pending: false,
+    //   accept: false,
+    // );
```
Comment (truyenjc) [inline lib/pages/friend_list/friend_list_page.dart]: remove

File: lib/widgets/button_widget.dart [hunk 13]
```diff
@@ -38,7 +38,7 @@ class ButtonWidget extends StatelessWidget {
             child: icon,
           ),
           FlatButton(
-            onPressed: _onPressed,
+            onPressed:_onPressed,
```
Comment (truyenjc) [inline lib/widgets/button_widget.dart]:
```suggestion
            onPressed: _onPressed,
```
format code

File: lib/pages/notification/widgets/new_friend_notice.dart [hunk 14]
```diff
@@ -0,0 +1,56 @@
...
+                  FlatButton(
+                    onPressed: onAcceptPressed,
+                    child: Text('Accept', style: TextStyle(color: Colors.blue)),
+                    textColor: Colors.black,
+                    shape: RoundedRectangleBorder(
+                        side: BorderSide(
+                            color: Colors.blue,
+                            width: 1,
+                            style: BorderStyle.solid),
+                        borderRadius: BorderRadius.circular(50)),
```
Comment (truyenjc) [inline lib/pages/notification/widgets/new_friend_notice.dart]:
```suggestion
                        borderRadius: BorderRadius.circular(50),),
```
format code

File: lib/pages/notification/detail_notice_page.dart [hunk 15]
```diff
@@ -0,0 +1,138 @@
...
+    Key key,
+    this.currentUserID,
+    this.currentUserName,
+  }) : super(key: key);
+
+  @override
+  _MyHomePageState createState() => _MyHomePageState();
+}
+
+class _MyHomePageState extends State<DetailNoticePage> {
```
Comment (truyenjc) [inline lib/pages/notification/detail_notice_page.dart]:
```suggestion
class _ DetailNoticePageState extends State<DetailNoticePage> {
```
(kill)

File: lib/pages/notification/detail_notice_page.dart [hunk 16]
```diff
@@ -0,0 +1,138 @@
...
+              return _onSuccessNotification(state.currentFriendsList);
+            }
+            return Loading();
+          },
+        ),
+      ),
+    );
+  }
+
+  Widget _onSuccessNotification(List<FriendRequest> listFriendRequest) {
```
Comment (truyenjc) [inline lib/pages/notification/detail_notice_page.dart]: function return a widget shouldn't use prefix ```on``` in name function

**Review by truyenjc — APPROVED** [review]
LGTM!

**Pull request merged**